### ISSUE TITLE
Add HTTP "Accept" header parsing

### ISSUE THREAD

**Issue by McSherry** [McSherry/Zener#36]
The HTTP `Accept` header is used by clients to specify the media types that are acceptable in a response. Including in Zener code to parse this header (which could utilise the `MediaType` class) would likely save time in a fair few scenarios.

Due to the format of `Accept` headers, they need some special handling.

```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
```

The inclusion of a `q` parameter to indicate preference means that some additional code is required to separate the parameter before passing the string to `MediaType.Create`, as otherwise `MediaType.Create` would either throw an exception or erroneously interpret the `q` parameter as a media type parameter.

**Comment by McSherry** [comment]
Completion of #38 partially completes this. The `OrderedCsvHttpHeader` class can be used to parse `Accept` headers. However, it would still be beneficial to abstract some of this. To complete this issue, the following should be implemented:

- [ ] An `IsAcceptable(this HttpRequest, MediaType)` extension method, which would compare the contents of the client's `Accept` header with the provided media type to determine whether the media type is acceptable.
- [ ] An `IsAcceptable(this HttpRequest, IEnumerable<MediaType>` extension method, which would determine which, if any, of the provided media types are acceptable. May also order them by preference, if available.

**Comment by McSherry** [comment]
Closed in 3ad6097.
